flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer09.java

```diff
@@ -181,12 +181,6 @@ public FlinkKafkaConsumer09(List<String> topics, KeyedDeserializationSchema<T> d
 
 		boolean useMetrics = !PropertiesUtil.getBoolean(properties, KEY_DISABLE_METRICS, false);
 
-		// make sure that auto commit is disabled when our offset commit mode is ON_CHECKPOINTS;
-		// this overwrites whatever setting the user configured in the properties
-		if (offsetCommitMode == OffsetCommitMode.ON_CHECKPOINTS || offsetCommitMode == OffsetCommitMode.DISABLED) {
-			properties.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
-		}
```

Comment:
This shouldn't be removed (I assume you accidentally removed it when rebasing?).